sockeye/data_io.py

```diff
@@ -245,17 +267,21 @@ class DataStatisticsAccumulator:
 
     def __init__(self,
                  buckets: List[Tuple[int, int]],
-                 vocab_source: Dict[str, int],
+                 vocab_source: Optional[Dict[str, int]],
                  vocab_target: Dict[str, int],
                  length_ratio_mean: float,
                  length_ratio_std: float) -> None:
         self.buckets = buckets
         num_buckets = len(buckets)
         self.length_ratio_mean = length_ratio_mean
         self.length_ratio_std = length_ratio_std
-        self.unk_id_source = vocab_source[C.UNK_SYMBOL]
+        if vocab_source is not None:
+            self.unk_id_source = vocab_source[C.UNK_SYMBOL]
+            self.size_vocab_source = len(vocab_source)
+        else:
+            self.unk_id_source = None
+            self.size_vocab_source = 0
```

Comment:
I am a bit confused about this comment. Do you mean have `self.vocab_size = 0` in the `PassThroughEmbeddingConfig, so that we can simplify the following code?
```
        w_embed_source = None
        if not isinstance(self.config.config_embed_source, encoder.PassThroughEmbeddingConfig):
            w_embed_source = mx.sym.Variable(prefix + C.SOURCE_EMBEDDING_PREFIX + "weight",
                                             shape=(self.config.config_embed_source.vocab_size,
                                                    self.config.config_embed_source.num_embed))
```

Reply:
for example. I was commenting about how we should represent the absence of a source vocabulary, either as 0 (aka an empty vocabulary) or None (aka no vocabulary at all). It would be good to have this consistent across all the changes in model & iterators etc.